tensorflow/core/kernels/cwise_ops_test.cc

```diff
@@ -29,7 +29,16 @@ static Graph* Unary(const string& func, int num, DataType dtype) {
   Graph* g = new Graph(OpRegistry::Global());
   Tensor data(dtype, TensorShape({64, 64, num / (64 * 64)}));
   CHECK_GT(data.NumElements(), 0);
-  data.flat<float>().setRandom();
+  switch (dtype) {
```

Comment:
I think it would be less error prone if we made this a template function of T -- then the C++ type could be part of the BM_UNARY macro below. You can see an example of this in the matmul benchmark:
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/kernels/matmul_op_test.cc#L23

https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/kernels/matmul_op_test.cc#L23